### PR TITLE
feat: add example of non-working global grpc_api_configuration file

### PR DESCRIPTION
**Pull request by gvencadze**
It's a reproducible example of issue https://github.com/grpc-ecosystem/grpc-gateway/issues/3059.

I've added two proto services, `grpc_configuration` and `globalconfig.buf.gen.yaml` files to `examples/internal/proto/globalconfig`

### CODE REVIEW

File: examples/internal/proto/globalconfig/grpc_annotations.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+type: google.api.Service
+config_version: 3
+
+http:
+  rules:
+      ### First service ###
+    - selector: api.first_svc.v1.FirstService.Get
```
Comment (johanbrandhorst) [inline examples/internal/proto/globalconfig/grpc_annotations.yaml:7]: This should match the package definition in the protobuf file
```suggestion
    - selector: grpc.gateway.examples.internal.proto.globalconfig.first_svc.v1.FirstService.Get
```

File: examples/internal/proto/globalconfig/grpc_annotations.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+      get: /api/v1/first/get
+
+      ### Second service ###
+    - selector: api.second_svc.v1.SecondService.Get
```
Comment (johanbrandhorst) [inline examples/internal/proto/globalconfig/grpc_annotations.yaml:11]: Same here

File: examples/internal/proto/globalconfig/globalconfig.buf.gen.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+version: v1
+plugins:
+  - name: grpc-gateway
+    out: .
+    opt:
+      - paths=source_relative
+      - grpc_api_configuration=examples/internal/proto/globalconfig/grpc_annotations.yaml
```
Comment (johanbrandhorst) [inline examples/internal/proto/globalconfig/globalconfig.buf.gen.yaml:7]: I don't think this option does anything to this generator?

**Review by johanbrandhorst — CHANGES_REQUESTED** [review]
I think you need to run the generation commands? See CONTRIBUTING.md in the root of the repository for a guide.

File: examples/internal/proto/globalconfig/globalconfig.buf.gen.yaml (hunk 3 above)
Comment (gvencadze) [reply examples/internal/proto/globalconfig/globalconfig.buf.gen.yaml:6]: Which option? `-paths=source-relative`?

**Comment by gvencadze** [comment]
@johanbrandhorst yeah, I've take a look at contributing guide, but I can't run regenerate command cause it raise following error:
```
Failure: plugin grpc-gateway: HTTP rules without a matching selector: .grpc.gateway.examples.internal.proto.globalconfig.second_svc.v1.SecondService.Get
HTTP rules without a matching selector: .grpc.gateway.examples.internal.proto.globalconfig.first_svc.v1.FirstService.Get
make: *** [Makefile:115: proto] Error 1
```

**Comment by gvencadze** [comment]
Btw, what do you think about ability to specify multiple configuration files? If it's okay, I'll take a look how to implement it

File: examples/internal/proto/globalconfig/globalconfig.buf.gen.yaml (hunk 3 above)
Comment (johanbrandhorst) [reply examples/internal/proto/globalconfig/globalconfig.buf.gen.yaml:6]: Huh, nevermind, it's here: https://github.com/grpc-ecosystem/grpc-gateway/blob/9b09466d8afe9003306397242ba3d1e3fbcd4cae/protoc-gen-grpc-gateway/main.go#L29. I don't know why I left this comment.

**Comment by johanbrandhorst** [comment]
> Btw, what do you think about ability to specify multiple configuration files? If it's okay, I'll take a look how to implement it

I really don't think that will be a good idea. I would sooner we allow users to specify multiple target files and directories in a single file. Specifying an external file is already a hack, unfortunately.

**Comment by johanbrandhorst** [comment]
> @johanbrandhorst yeah, I've take a look at contributing guide, but I can't run regenerate command cause it raise following error:
> 
> ```
> Failure: plugin grpc-gateway: HTTP rules without a matching selector: .grpc.gateway.examples.internal.proto.globalconfig.second_svc.v1.SecondService.Get
> HTTP rules without a matching selector: .grpc.gateway.examples.internal.proto.globalconfig.first_svc.v1.FirstService.Get
> make: *** [Makefile:115: proto] Error 1
> ```

Yeah uhm I'm not entirely sure what the selector _should_ be. Try and play around with it a bit 😝

**Comment by gvencadze** [comment]
@johanbrandhorst selector is right, If I delete second_svc rule and remove folder "second_svc" everything generates as expected, otherwise returns selector error. Also it's look kinda strange that in output with error selector is got dot at the start of selector e.g. instead of `app_selector` in output I see `.app_selector`

**Comment by gvencadze** [comment]
> I really don't think that will be a good idea. I would sooner we allow users to specify multiple target files and directories in a single file. Specifying an external file is already a hack, unfortunately.

Oh, if it's already a hack, than adding new features to it will be strange, I agree

**Comment by johanbrandhorst** [comment]
> @johanbrandhorst selector is right, If I delete second_svc rule and remove folder "second_svc" everything generates as expected, otherwise returns selector error. Also it's look kinda strange that in output with selector error we got dot at the start of selector e.g. instead of `app_selector` we got `.app_selector`

It does look strange, I agree. I think it has to do with how the protobuf internal representation looks. I'm not sure why you're getting that selector error, you might have to do some debugging in the code, sorry. This part of the generator is rarely used and could do with some love.

**Comment by gvencadze** [comment]
> It does look strange, I agree. I think it has to do with how the protobuf internal representation looks. I'm not sure why you're getting that selector error, you might have to do some debugging in the code, sorry. This part of the generator is rarely used and could do with some love.

Maybe you can share any instruction how to debug such instruments? I tried run it as standard go application, but got error saying that it should be run using protoc

**Pull request closed without merging**